### PR TITLE
Update Schematron Model and Processor to use new/updated Schematron Error Messages from APHL

### LINKED ISSUE #623: Fix Schematron Error Texts in our schematron model and processor

### PR DESCRIPTION
**Pull request by BradySkylight**
## Description

During testing APHL was able to find that many eICRs that should have been processed by TTC did not and were marked as 'No Schematron Errors to process' essentially.  This was due to a mis-match in the Schematron Error Messages between what was being used in APHL's pipeline and the TTC code base.

We received new Schematron files with updated ids and messages.  I have updated the Schematron Model to use the new/update Schematron Error Messages.  Tests were updated as well.  We will have another ticket/PR to change from using the Error Message to using the AssertionID in the near future (this is already a WIP).

## Related Issues

Closes #623 

## Checklist for Reviewers

Please review and complete the following checklist during the review process:

- [ ] The code follows best practices and conventions.
- [ ] The changes implement the desired functionality or fix the reported issue.
- [ ] The tests cover the new changes and pass successfully.
- [ ] Any potential edge cases or error scenarios have been considered.

### CODE REVIEW

File: packages/text-to-code/src/text_to_code/models/schematron.py [hunk 1]
```diff
@@ -7,15 +7,21 @@
 class LabTestNameOrderedSchematronErrors(Enum):
     """The list of Schematron error messages relevant to the lab test name ordered data field."""
 
-    MISSING_CODE_ATTRIBUTE = "Text to Code: Lab Test Name Ordered does not have a @code attribute"
-    INVALID_CODE_SYSTEM = "Text to Code: Lab Test Name Ordered code and translation data elements @codeSystem attribute are not LOINC 2.16.840.1.113883.6.1"
+    MISSING_CODE_ATTRIBUTE = (
+        "Text to Code: Planned observation code data element has no @code attribute"
+    )
+    INVALID_CODE_SYSTEM = "Text to Code: Lab Test Name Resulted code and translation data elements @codeSystem attribute are not LOINC (2.16.840.1.113883.6.1), SNOMED CT (2.16.840.1.113883.6.96),  CPT-4 (2.16.840.1.113883.6.12), ICD10 PCS (2.16.840.1.113883.6.4) or CDT-2 (2.16.840.1.113883.6.13)"
```
Comment (nickclyde) [inline packages/text-to-code/src/text_to_code/models/schematron.py:13]: Just double checking: this message says "Lab Test Name Resulted" but it's in the Ordered enum. Is this a mistake on APHL's part?

File: packages/validation/src/validation/eicr-validator/schematron/APHL_TextToCodeSchematron_09252025.sch [hunk 2]
```diff
@@ -25,7 +25,7 @@
       <sch:assert id="ttc-labTestNameOrdered-noCode" test="
         not(cda:code) 
         or cda:code/@code
-        or cda:code/cda:translation/@code">Text to Code: Lab Test Name Ordered does not have a @code attribute</sch:assert>
+        or cda:code/cda:translation/@code">Text to Code: Planned observation code data element has no @code attribute</sch:assert>
```
Comment (nickclyde) [inline packages/validation/src/validation/eicr-validator/schematron/APHL_TextToCodeSchematron_09252025.sch:28]: It looks like only the two noCode messages were updated here, so is this file now partially out of sync with the new enums? The `ttc-labTestNameOrdered-wrongCodeSystem` assert below still says "Lab Test Name Ordered @codeSystem attribute is not LOINC...", which doesn't match the old or the new `INVALID_CODE_SYSTEM` enum value, and the new blank/nullFlavor asserts don't exist in this file at all?

Could we just drop in the full new `.sch` that APHL sent here?

File: e2e/snapshots/test_e2e/test_upload_and_process/augmentation_metadata.json [hunk 3]
```diff
@@ -1 +1,22 @@
-{"original_eicr_id": "c8516bdc-8bb2-40aa-8dae-20a77546488f", "augmented_eicr_id": "d44dc1c6-8a0c-5236-906e-12f6475589ec", "nonstandard_codes": [{"schematron_error": "Text to Code: Lab Test Name Ordered does not have a @code attribute", "schematron_error_xpath": "\n                    /ClinicalDocument/component[1]/structuredBody[1]/component[1]/section[1]/entry[1]/observation[1]", "field_type": "Lab Test Name Ordered", "new_translation": {"code": "82041-5", "code_system": "2.16.840.1.113883.6.1", "code_system_name": "LOINC", "display_name": "Weed Allerg Mix3 IgE Qn", "value_set": null, "value_set_version": null, "original_text": "A custom code in display name."}, "new_translation_xpath": "/ClinicalDocument/component/structuredBody/component/section/entry/observation/code/translation"}], "error": null}
\ No newline at end of file
+{
```
Comment (nickclyde) [inline e2e/snapshots/test_e2e/test_upload_and_process/augmentation_metadata.json:1]: Is this snapshot still used? I vaguely remember the e2e tests changing the way they snapshot when we did the `passthrough_reason` work.

File: packages/text-to-code/tests/unit/test_schematron_processor.py [hunk 4]
```diff
@@ -92,18 +92,18 @@ def test_get_schematron_error_detail_fields(self):
             for error in error_result
             if error.field == DataField.LAB_TEST_NAME_RESULTED
             and error.error_message
-            == "Text to Code: Lab Test Name Resulted does not have a @code attribute"
+            == "Text to Code: Lab Test Name Resulted does not have a @code attribute."
```
Comment (robertandremitchell) [inline packages/text-to-code/tests/unit/test_schematron_processor.py]: should we just hardcode this too, given how often it pops up / our concern about potential changes

**Review by robertandremitchell — APPROVED** [review]
lgtm, one small note about one of the tests

File: packages/text-to-code/src/text_to_code/models/schematron.py [hunk 5]
```diff
@@ -7,15 +7,21 @@
 class LabTestNameOrderedSchematronErrors(Enum):
     """The list of Schematron error messages relevant to the lab test name ordered data field."""
 
-    MISSING_CODE_ATTRIBUTE = "Text to Code: Lab Test Name Ordered does not have a @code attribute"
-    INVALID_CODE_SYSTEM = "Text to Code: Lab Test Name Ordered code and translation data elements @codeSystem attribute are not LOINC 2.16.840.1.113883.6.1"
+    MISSING_CODE_ATTRIBUTE = (
```
Comment (JNygaard-Skylight) [inline packages/text-to-code/src/text_to_code/models/schematron.py:10]: Will we change TTC to match on the issue ID in a different PR?

File: packages/text-to-code/src/text_to_code/models/schematron.py (hunk 1 above)
Comment (BradySkylight) [reply packages/text-to-code/src/text_to_code/models/schematron.py:13]: That's an APHL error.  That is the text we have in all areas.

File: packages/validation/src/validation/eicr-validator/schematron/APHL_TextToCodeSchematron_09252025.sch (hunk 2 above)
Comment (BradySkylight) [reply packages/validation/src/validation/eicr-validator/schematron/APHL_TextToCodeSchematron_09252025.sch:28]: I am waiting to get a new file from Tim.  I requested he send me a new 'stitched' file that has the updated and new error messages in it.  Once we have that, I'll add it, rename it, and ensure we are using that in our validation process.  GOOD CATCH and call-out here.

File: e2e/snapshots/test_e2e/test_upload_and_process/augmentation_metadata.json (hunk 3 above)
Comment (BradySkylight) [reply e2e/snapshots/test_e2e/test_upload_and_process/augmentation_metadata.json:1]: I am not sure actually.  Happy to remove if it's not being used.  @joshnygaard or @robertandremitchell do you know?

File: packages/text-to-code/src/text_to_code/models/schematron.py (hunk 5 above)
Comment (BradySkylight) [reply packages/text-to-code/src/text_to_code/models/schematron.py:10]: Yes, I have a draft PR out that has us about 80% of the way there.  ;)

File: packages/text-to-code/tests/unit/test_schematron_processor.py (hunk 4 above)
Comment (BradySkylight) [reply packages/text-to-code/tests/unit/test_schematron_processor.py]: I'm not sure I'm following?  Are you meaning: maybe have a variable set in the file that has this value and use it over and over again instead of having it written out like 6 times in the same file?

Comment (robertandremitchell) [reply packages/text-to-code/tests/unit/test_schematron_processor.py]: Yes, sorry, that, just making it a constant since this text might change again before we switch to using codes.

Comment (BradySkylight) [reply packages/text-to-code/tests/unit/test_schematron_processor.py]: Agreed.  I'm on it and will fix and push.

File: e2e/snapshots/test_e2e/test_upload_and_process/augmentation_metadata.json (hunk 3 above)
Comment (robertandremitchell) [reply e2e/snapshots/test_e2e/test_upload_and_process/augmentation_metadata.json:1]: it's still generated, but I think we will need to probably revisit how the e2e works. Tackling some of that in my current ticket on persistence ID, so I could look at that there so as to not block this work.

Comment (BradySkylight) [reply e2e/snapshots/test_e2e/test_upload_and_process/augmentation_metadata.json:1]: Thanks Rob!!

File: packages/validation/src/validation/eicr-validator/schematron/APHL_TextToCodeSchematron_09252025.sch (hunk 2 above)
Comment (BradySkylight) [reply packages/validation/src/validation/eicr-validator/schematron/APHL_TextToCodeSchematron_09252025.sch:28]: Am I ok to 'resolve' this comment for now @nickclyde, as it seems no tests are failing currently.  I'll open a new PR with that file updated and leveraged.  Does that work?

**Review by nickclyde — APPROVED** [review]
